runtime/vm/resolvesupport.cpp

```diff
@@ -1851,6 +1851,122 @@ resolveMethodHandleRef(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpInde
 
 }
 
+j9object_t
+resolveConstantDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpIndex, UDATA resolveFlags)
+{
+	J9RAMConstantDynamicRef *ramCPEntry = (J9RAMConstantDynamicRef*)ramCP + cpIndex;
+	j9object_t value = NULL;
+	J9JavaVM *vm = vmThread->javaVM;
+	bool resolved = false;
+
+retry:
+	/* Fast path resolution for previously resolved entries */
+	if (NULL != ramCPEntry->value) {
+		value = ramCPEntry->value;
+		resolved = true;
+	} else {
+		/* check if already resolved to NULL or exception */
+		if (NULL != ramCPEntry->exception) {
+			J9Class *throwable = J9VMJAVALANGTHROWABLE_OR_NULL(vm);
+			J9Class *clazz = J9OBJECT_CLAZZ(vmThread, ramCPEntry->exception);
+			if (ramCPEntry->exception == vm->voidReflectClass->classObject) {
+				/* Resolved with null reference */
+				resolved = true;
+			} else if (isSameOrSuperClassOf(throwable, clazz)) {
+				/* Resolved with exception */
+				VM_VMHelpers::setExceptionPending(vmThread, ramCPEntry->exception);
+				resolved = true;
+			}
+		}
+	}
+
+	if (resolved) {
+		return value;
```

Comment:
Multiple returns are discouraged.  Please use if (!resolved) instead of if/return/else.

Reply:
fixed